aws_lambda_powertools/utilities/parameters/__init__.py

```diff
@@ -0,0 +1,247 @@
+    """
+
+    client = None
+
+    def __init__(
+        self, region: Optional[str] = None,
+    ):
+        """
+        Initialize the SSM Parameter Store client
+        """
+
+        client_kwargs = {}
+        if region:
+            client_kwargs["region_name"] = region
+
+        self.client = boto3.client("ssm", **client_kwargs)
+
+        super().__init__()
+
+    def _get(self, name: str, **kwargs) -> str:
+        """
+        Retrieve a parameter value from AWS Systems Manager Parameter Store
+
+        Parameters
+        ----------
+        name: str
+            Parameter name
+        decrypt: bool
+            If the parameter value should be decrypted
+        """
+
+        # Load kwargs
+        decrypt = kwargs.get("decrypt", False)
+
+        return self.client.get_parameter(Name=name, WithDecryption=decrypt)["Parameter"]["Value"]
+
+    def _get_multiple(self, path: str, **kwargs) -> Dict[str, str]:
+        """
+        Retrieve multiple parameter values from AWS Systems Manager Parameter Store
+
+        Parameters
+        ----------
+        path: str
+            Path to retrieve the parameters
+        decrypt: bool
+            If the parameter values should be decrypted
+        recursive: bool
+            If this should retrieve the parameter values recursively or not
+        """
+
+        # Load kwargs
+        decrypt = kwargs.get("decrypt", False)
+        recursive = kwargs.get("recursive", False)
+
+        response = self.client.get_parameters_by_path(Path=path, WithDecryption=decrypt, Recursive=recursive)
+        parameters = response.get("Parameters", [])
+
+        # Keep retrieving parameters
+        while "NextToken" in response:
+            response = self.client.get_parameters_by_path(
+                Path=path, WithDecryption=decrypt, Recursive=recursive, NextToken=response["NextToken"]
+            )
+            parameters.extend(response.get("Parameters", []))
+
+        retval = {}
+        for parameter in parameters:
+
+            # Standardize the parameter name
+            # The parameter name returned by SSM will contained the full path.
+            # However, for readability, we should return only the part after
+            # the path.
+            name = parameter["Name"]
+            if name.startswith(path):
+                name = name[len(path) :]
+            name = name.lstrip("/")
+
+            retval[name] = parameter["Value"]
+
+        return retval
+
+
+class SecretsProvider(BaseProvider):
+    """
+    AWS Secrets Manager Parameter Provider
+    """
+
+    client = None
+
+    def __init__(self, region: Optional[str] = None):
+        """
+        Initialize the Secrets Manager client
+        """
+
+        client_kwargs = {}
+        if region:
+            client_kwargs["region_name"] = region
+
+        self.client = boto3.client("secretsmanager", **client_kwargs)
+
+        super().__init__()
+
+    def _get(self, name: str, **kwargs) -> str:
```

Comment:
What do you think of standardizing on on additional options via a `sdk_options` keyword param?

If no params are passed we're fine as the call works, otherwise, it still gives them the flexibility to pass in things like [VersionId, VersionStage](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager.html#SecretsManager.Client.get_secret_value)

This would change to

`return self.client.get_secret_value(SecretId=name, **sdk_options)["SecretString"]`